cluster.tf

```diff
@@ -1,104 +1,72 @@
-      subnet_id = zones.value["id"]
-      name      = zones.value["zone"]
-    }
-  }
-
-  dynamic "kms_config" {
-    for_each = each.value.kms_config == null ? [] : [each.value.kms_config]
-    content {
-      crk_id           = module.key_management.key_map[kms_config.value.crk_name].key_id
-      instance_id      = module.key_management.key_management_guid
-      private_endpoint = kms_config.value.private_endpoint
-    }
-  }
-
-  disable_public_service_endpoint = true
-
-  timeouts {
-    create = "3h"
-    delete = "2h"
-    update = "3h"
-  }
-
-}
-
-##############################################################################
-
-
-##############################################################################
-# Create Worker Pools
-##############################################################################
-
-resource "ibm_container_vpc_worker_pool" "pool" {
-  for_each          = local.worker_pools_map
+  region            = var.region
+  cluster_name      = each.value.cluster_name
   vpc_id            = each.value.vpc_id
-  resource_group_id = local.resource_groups[each.value.resource_group]
-  entitlement       = each.value.entitlement
-  cluster           = ibm_container_vpc_cluster.cluster[each.value.cluster_name].id
-  worker_pool_name  = each.value.name
-  flavor            = each.value.flavor
-  worker_count      = each.value.workers_per_subnet
-
-  dynamic "zones" {
-    for_each = each.value.subnets
-    content {
-      subnet_id = zones.value["id"]
-      name      = zones.value["zone"]
-    }
+  ocp_entitlement   = each.value.entitlement
+  vpc_subnets = {
+    vsi-zone-1 = [
+      for zone in each.value.subnets :
+      {
+        id         = zone.id
+        zone       = zone.zone
+        cidr_block = zone.cidr
+      }
+    ]
+  }
+  worker_pools = concat(
+    [
+      {
+        subnet_prefix    = each.value.subnet_names[0]
+        pool_name        = "default"
+        machine_type     = each.value.machine_type
+        workers_per_zone = each.value.workers_per_subnet
+        boot_volume_encryption_kms_config = {
+          crk             = module.key_management.key_map[each.value.kms_config.crk_name].key_id
+          kms_instance_id = module.key_management.key_management_guid
+        }
+      }
+    ],
+    [
+      for pool in each.value.worker_pools :
+      {
+        subnet_prefix    = pool.subnet_names[0]
+        pool_name        = pool.name
+        machine_type     = pool.flavor
+        workers_per_zone = pool.workers_per_subnet
+        boot_volume_encryption_kms_config = {
+          crk             = module.key_management.key_map[each.value.kms_config.crk_name].key_id
+          kms_instance_id = module.key_management.key_management_guid
+        }
+      }
+  ])
+  ocp_version                     = each.value.kube_version
+  tags                            = var.resource_tags
+  use_existing_cos                = true
+  disable_public_endpoint         = each.value.disable_public_endpoint
```

Comment:
You need to look this up and then either set it if it has a value or default it to "true"